WINGSTOP_Air_Brake_Flight_Computer_FULL_BRAKES/WINGSTOP_Air_Brake_Flight_Computer_FULL_BRAKES.ino

```diff
@@ -373,70 +416,71 @@ void read_barometer(){//takes barometer reading
 }
 
 float read_altitude(){//reads and returns barometer altitude reading
-  while (! bmp.performReading()) {
-    //performs until it gets a valid reading
-  }
-  return bmp.readAltitude(init_pressure); // returns current altitude reading
+  float altitudeReading;
+  do {
+    read_barometer(); 
+    altitudeReading = bmp.readAltitude(init_pressure);
+  } while (abs(altitudeReading - x_current)/delta_t > 1200); 
+  //keeps reading altimiter if vertical speed is more than 1200 m/s (should never actually be that fast).
+  return altitudeReading; // returns current altitude reading
 }
 
 
 //***********BRAKE FUNCTIONS***********
 
 void open_brakes(){//opens brakes fully
-  brakes.write(brakes_open);//600 to 1580 for full range
+  brakes.write(brakes_open);
 }
 
 void close_brakes(){//opens brakes fully
-  brakes.write(brakes_closed);//600 to 1580 for full range
+  brakes.write(brakes_closed);
 }
 
-void set_brakes(float pos){//set brakes to any position between 0 and 1 (0 = closed, 1 = open) 
-  brakes.write((int)(pos*(brakes_open - brakes_closed) + brakes_closed));//600 to 1580 for full range
+void set_brakes(){//set brakes to desired brake position between 0 and 100 (0 = closed, 100 = open) 
```

Comment:
This function should take a parameter `pos` for the position, rather than using a global variable.  It makes it much clearer what's being set.  Global variables should be used sparingly.